packages/bot/src/scratch/utils/scratch-helper.js

```diff
@@ -0,0 +1,22 @@
+import config                   from '../../constants';
+
+export const hasAllRequiredBlocks = () => {
+    const blocks_in_workspace     = Blockly.derivWorkspace.getAllBlocks();
+    const { mandatoryMainBlocks } = config;
+    const required_block_types    = ['trade_definition_tradeoptions', ...mandatoryMainBlocks];
+    const all_block_types         = blocks_in_workspace.map(block => block.type);
+    const has_all_required_blocks = required_block_types.every(required_block_type =>
+        all_block_types.includes(required_block_type));
+
+    return has_all_required_blocks;
+};
+
+export const onWorkspaceResize = () => {
+    const el_scratch_div  = document.getElementById('scratch_div');
+
+    if (el_scratch_div) {
+        el_scratch_div.style.width = 'var(--bot-content-width)';
+        el_scratch_div.style.height =  'var(--bot-content-height)';
```

Comment:
- Is this necessary considering we're updating the value of that CSS variable later on (in `packages/bot/src/utils/window-size.js`)? If I'm not mistaken, the browser would detect those changes and automatically apply the new values.
- `Blockly.svgResize` may still need to be executed.

Reply:
@aaron-binary it it cause this also change the size on windows resize. we have a listener for it. we only change css-var value on run-panel open and close.
im not sure what this do `Blockly.svgResize`